crates/cxx-qt-lib-extras/include/core/qlocale.h

```diff
@@ -0,0 +1,31 @@
+// clang-format off
+// SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// clang-format on
+// SPDX-FileContributor: Nicolas Fella <nicolas.fella@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+#pragma once
+
+#include <QtCore/QLocale>
+
+#include "rust/cxx.h"
+
+namespace rust {
+
+template<>
+struct IsRelocatable<QLocale> : ::std::true_type
+{};
+
+namespace cxxqtlib1 {
+
+using QLocaleTagSeparator = QLocale::TagSeparator;
+using QLocaleCurrencySymbolFormat = QLocale::CurrencySymbolFormat;
+using QLocaleFormatType = QLocale::FormatType;
+using QLocaleLanguage = QLocale::Language;
+using QLocaleMeasurementSystem = QLocale::MeasurementSystem;
+
+QLocale fromName(const QString& name);
+
+QLocale defaultLocale();
```

Comment:
you likely don't need these and can use our constructor helper from cxx-qt-lib/common.h

eg

```rust
#[namespace = "rust::cxxqtlib1"]
unsafe extern "C++ {
   include!("cxx-qt-lib/common.h");
  
   #[doc(hidden)]
   #[rust_name = "qlocale_init_default"]
   fn construct() -> QLocale;

   #[doc(hidden)]
   #[rust_name = "qlocale_init_from_name"]
   fn construct(name: &QString) -> QLocale;
}
```

This would likely work in your CXX bridge then you can do `ffi::qlocale_init_from_name(my_string)` and `ffi::qlocale_init_default()`